.github/workflows/release.yml

```diff
@@ -0,0 +1,45 @@
+name: release
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - 'master'
+    paths:
+      - 'usr/local/share/bastille/version.sh'
+
+jobs:
+  release:
+    permissions:
+      contents: write
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: bastille-version
+        shell: bash
```

Comment:
`runs-on` could be `macos-latest`? Not sure if that will need to change the shell to `sh`, but is just to honor real unix and freebsd legacy.

Regards :)

Reply:
:facepalm: great catch! That's a tough one.

I'm now hesitant: 

On one hand if this PR is the first step towards more automation, there might be a need for deeper discussion on where/how CI should happen. The obvious solution is to run our own runner, but that's actually [not recommended by GH for public repos](https://docs.github.com/en/actions/hosting-your-own-runners/managing-self-hosted-runners/adding-self-hosted-runners) and has the disadvantage that someone in the community would have to maintain the runner. We could also explore https://github.com/marketplace/actions/freebsd-vm to still run on GitHub. Also for comparison Poudriere uses Cirrus CI, and incidentally autotools for package versioning.

On the other hand, if it's just for version release automation, this PR works ok as is ( :crossed_fingers: ). I agree `macos-latest` is probably slightly more suited.

Reply:
By now this should do the version thing only. We should think in CI and testing later of course for GitHub actions, but at this point if it does the version thing correctly then it will be ok if uses `macos-latest` if possible.

Anyway we will need to discuss this with @cedwards and test it before doing any merge.